PowerEditor/src/NppCommands.cpp

```diff
@@ -105,6 +105,15 @@ void Notepad_plus::command(int id)
 		}
 		break;
 		
+		case IDM_FILE_OPEN_DEFAULT_VIEWER:
+		{
+			// Opens file in its default viewer. 
+            // Has the same effect as double–clicking this file in Windows Explorer.
+            BufferID buf = _pEditView->getCurrentBufferID();
+			ShellExecute(NULL, NULL, buf->getFullPathName(), NULL, NULL, SW_SHOW);

```

Comment:
What if there is no program associated with a particular file type. For example: one might have python script file (.py extension), then?

IMO, the "Open in Default Viewer" (under tab and File Menu) should be enabled only when a program/executable is associated with a particular file type.